typescript/sdk/src/token/deploy.ts

```diff
@@ -131,91 +170,20 @@ export class HypERC20Deployer extends GasRouterDeployer<
     } as ERC20Metadata;
   }
 
-  protected async deployCollateral(
-    chain: ChainName,
-    config: HypERC20CollateralConfig,
-  ): Promise<HypERC20Collateral> {
-    let contractName:
-      | TokenType.fastCollateral
-      | TokenType.collateral
-      | TokenType.collateralVault;
-    switch (config.type) {
-      case TokenType.fastSynthetic || TokenType.fastCollateral:
-        contractName = TokenType.fastCollateral;
-        break;
-      case TokenType.collateral:
-        contractName = TokenType.collateral;
-        break;
-      case TokenType.collateralVault:
-        contractName = TokenType.collateralVault;
-        break;
-      default:
-        throw new Error(`Unknown collateral type ${config.type}`);
-    }
-    return this.deployContract(chain, contractName, [
-      config.token,
-      config.mailbox,
-    ]);
-  }
-
-  protected async deployNative(
-    chain: ChainName,
-    config: HypNativeConfig,
-  ): Promise<HypNative> {
-    if (config.scale) {
-      return this.deployContract(chain, TokenType.nativeScaled, [
-        config.scale,
-        config.mailbox,
-      ]);
-    } else {
-      return this.deployContract(chain, TokenType.native, [config.mailbox]);
-    }
-  }
-
-  protected async deploySynthetic(
-    chain: ChainName,
-    config: HypERC20Config,
-  ): Promise<HypERC20> {
-    const router: HypERC20 = await this.deployContract(
-      chain,
-      isFastConfig(config) ? TokenType.fastSynthetic : TokenType.synthetic,
-      [config.decimals, config.mailbox],
-    );
-    try {
-      await this.multiProvider.handleTx(
-        chain,
-        router.initialize(config.totalSupply, config.name, config.symbol),
-      );
-    } catch (e: any) {
-      if (!e.message.includes('already initialized')) {
-        throw e;
-      }
-      this.logger(`${config.type} already initialized`);
-    }
-    return router;
-  }
-
   router(contracts: HyperlaneContracts<HypERC20Factories>) {
     for (const key of objKeys(hypERC20factories)) {
       if (contracts[key]) {
-        return contracts[key];
+        return contracts[key] as GasRouter;
       }
     }
     throw new Error('No matching contract found');
   }
```

Comment:
maybe this should take the config too? and then we can do `contracts[config.tokenType]`

Reply:
Same as above. It's overriding the abstract so keeping the same sig.